src/main/java/com/conveyal/file/LocalFileStorage.java

```diff
@@ -62,7 +61,15 @@ public void moveIntoStorage(FileStorageKey key, File sourceFile) {
                         sourceFile.getName());
             }
             // Set the file to be read-only and accessible only by the current user.
-            Files.setPosixFilePermissions(storedFile.toPath(), Set.of(OWNER_READ));
+            try {
+                Files.setPosixFilePermissions(storedFile.toPath(), Set.of(OWNER_READ));
+            } catch (UnsupportedOperationException e) {
+                LOG.info("Could not restrict permissions on {} to POSIX OWNER_READ (probably because OS is not POSIX " +
+                        "compatible)", sourceFile.getName());
+                if (storedFile.setReadOnly()) LOG.info("Marked {} read-only via alternative setReadOnly() method",
```

Comment:
We should probably just call `storedFile.setReadOnly()` in all cases, and remove the call to `setPosixFilePermissions`. It looks like setReadOnly will adapt to different platforms via the `FileSystem` abstraction.

Reply:
The javadoc does not make it clear whether setReadOnly applies to only the user or everyone. However there is a setWritable (bool, bool) method that allows you to explicitly specify it should not be writable by anyone.